src/auth/googleauth.ts

```diff
@@ -615,4 +651,64 @@ export class GoogleAuth {
 
     return {client_email: data.default.email};
   }
+
+  /**
+   * Automatically obtain a client based on the provided configuration.  If no
+   * options were passed, use Application Default Credentials.
+   */
+  async getClient() {
+    if (!this.cachedCredential) {
+      if (this.keyfileName) {
+        const filePath = path.resolve(process.cwd(), this.keyfileName);
+        console.log(filePath);
+        const stream = fs.createReadStream(filePath);
+        this.cachedCredential = await this.fromStreamAsync(stream);
+      } else if (this.jsonContent) {
+        this.cachedCredential = await this.fromJSON(this.jsonContent);
+      } else {
+        await this.getApplicationDefaultAsync();
```

Comment:
no assignment on this path?

Reply:
It's a little gross, but that method actually caches the result itself so the assignment isn't needed.